src/imp.rs

```diff
@@ -49,6 +50,15 @@ pub enum Error<S: Store, SC: StateClient, PC: ProverClient> {
     /// Note combination for the given value is impossible given the maximum
     /// amount if inputs in a transaction.
     NoteCombinationProblem,
+    /// The key is already staked. This happens when there already is an amount
+    /// staked for a key and the user tries to make a stake transaction.
+    AlreadyStaked((PublicKey, StakeInfo)),
```

Comment:
Maybe instead of tuples we want here a struct? Example:

```
AlreadyStaked {
  key: PublicKey,
  stake: StakeInfo,
}
```

Reply:
Make sense! Will change.